ee/hogai/graph/root/nodes.py

```diff
@@ -299,6 +303,16 @@ class RootNode(RootNodeUIContextMixin):
     """
     CONVERSATION_WINDOW_SIZE = 64000
 
+    async def get_reasoning_message(
```

Comment:
Question: Do we scan the whole conversation with `find_last_ui_context`? Do we care?